src/google/protobuf/compiler/cpp/cpp_helpers.cc

```diff
@@ -1011,29 +1011,16 @@ bool IsWellKnownMessage(const FileDescriptor* file) {
   return well_known_files.find(file->name()) != well_known_files.end();
 }
 
-static bool FieldEnforceUtf8(const FieldDescriptor* field,
```

Comment:
Sorry but unfortunately we need to keep these functions around. Within our internal version of the codebase we have some additional code in these functions for legacy reasons, to allow the UTF-8 enforcement to be disabled for particular fields. Even though the functions look pointless, it helps us to have them around so that we can easily apply our internal changes.

If you just want to fix an unused-parameter warning, it would be fine to remove the parameter names (i.e. just make it `(const FieldDescriptor*, const Options&)`) or add lines like `(void) field;`.

Reply:
I am not working on unused parameter in protoc itself, but rather in the generated code.

It is ok if these methods would not be removed, though this will make interfaces a bit clumsier.

Could you, please, take a look on the question below?